crates/rust-analyzer/src/bin/logger.rs

```diff
@@ -38,14 +40,13 @@ impl<'a> MakeWriter<'a> for MakeWriterStderr {
     }
 }
 
-impl Logger {
-    pub(crate) fn new(file: Option<File>, filter: Option<&str>) -> Logger {
-        let filter = filter.map_or(EnvFilter::default(), EnvFilter::new);
+impl LoggerConfig {
+    pub(crate) fn init(self) -> anyhow::Result<()> {
+        let mut filter: Targets = self
+            .filter
+            .parse()
+            .with_context(|| format!("invalid log filter: `{}`", self.filter))?;
 
-        Logger { filter, file }
-    }
-
-    pub(crate) fn install(self) -> Result<()> {
         // The meaning of CHALK_DEBUG I suspected is to tell chalk crates
```

Comment:
The comment is a bit confusing now that the CHALK_DEBUG var is being read from somewhere else